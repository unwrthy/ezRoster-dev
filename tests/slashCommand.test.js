const slashCommandFactory = require('../src/slashCommand')

jest.mock('../src/commandParser', () => (text) => {
  if (!text) {
    return {
      urls: undefined,
      domain: undefined,
      slashtags: undefined
    }
  }

  return {
    urls: ['http://example.com'],
    domain: 'somedomain.co',
    slashtags: 'test'
  }
})

test('It should return a function', () => {
  const createShortUrls = jest.fn()
  expect(typeof slashCommandFactory(createShortUrls, 'someSlackToken')).toBe('function')
})

test('It should reject if no body is provided', () => {
  const createShortUrls = jest.fn()
  const slashCommand = slashCommandFactory(createShortUrls, 'someSlackToken')
  expect(slashCommand()).resolves.toMatchSnapshot()
})

test('It should reject if the wrong token is provided', () => {
  const createShortUrls = jest.fn()
  const slashCommand = slashCommandFactory(createShortUrls, 'someSlackToken')
  expect(slashCommand({token: 'wrongToken'})).resolves.toMatchSnapshot()
})

test('It should reject if the command input is not valid', () => {
  const createShortUrls = jest.fn()
  const slashCommand = slashCommandFactory(createShortUrls, 'someSlackToken')
  expect(slashCommand({token: 'someSlackToken'})).resolves.toMatchSnapshot()
})

test('It should create a proper slack response for short urls', () => {
  const sampleResult = require('./fixtures/sampleShortUrlsResult')
  const createShortUrls = jest.fn().mockReturnValueOnce(Promise.resolve(sampleResult))
  const slashCommand = slashCommandFactory(createShortUrls, 'someSlackToken')
  expect(slashCommand({token: 'someSlackToken', text: 'https://facebook.github.io/jest/docs/expect.html'})).resolves.toMatchSnapshot()
})
