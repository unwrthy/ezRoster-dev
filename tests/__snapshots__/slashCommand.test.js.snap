// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should create a proper slack response for short urls 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "good",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "*<http://rebrand.ly/dxws|rebrand.ly/dxws>* (<https://www.rebrandly.com/links/1611678|edit>):
https://facebook.github.io/jest/docs/expect.html",
    },
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "*Error*:
Sample error",
    },
  ],
  "text": "2 link(s) processed",
}
`;

exports[`It should reject if no body is provided 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "*Error*:
Invalid body",
    },
  ],
  "text": "",
}
`;

exports[`It should reject if the command input is not valid 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "*Error*:
No url found in the message",
    },
  ],
  "text": "",
}
`;

exports[`It should reject if the wrong token is provided 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "*Error*:
Invalid token",
    },
  ],
  "text": "",
}
`;
